### PR TITLE
TFLite: transpose_conv: fix padding height/width

### PR DESCRIPTION
**Pull request by perfinion**
The Conv2D_transpose op in TFLite does not handle non-square padding
correctly and the outputs do not match regular TensorFlow. Since this op
is transposed, height and width of the padding must be swapped.
Thanks to Matt Elsey for discovering this and developing the fix.

Signed-off-by: Jason Zaman <jason@perfinion.com>

This verifies the error:

```
#!/usr/bin/env python3

import tensorflow as tf
import numpy as np
import matplotlib
matplotlib.use("agg")
from matplotlib import pyplot as plt


IN_IMG_SHAPE = (1, 50, 54, 3)
IN_FILTER_SHAPE = (3, 3, 6, 3)
OUT_SHAPE = (1, 100, 108, 8)

errs = np.zeros((10, 10))

for xf in range(10):
    for yf in range(10):
        IN_FILTER_SHAPE = (1+xf, 1+yf, 8, 3)

        xx = np.array(np.random.random_sample(IN_IMG_SHAPE), dtype=np.float32)
        kk = np.array(np.random.random_sample(IN_FILTER_SHAPE), dtype=np.float32)

        x = tf.placeholder(dtype=tf.float32, shape=IN_IMG_SHAPE)
        k = tf.placeholder(dtype=tf.float32, shape=IN_FILTER_SHAPE)

        y = tf.nn.conv2d_transpose(x, k, output_shape=OUT_SHAPE, padding='SAME', strides=(1, 2, 2, 1))


        with tf.Session() as sess:
            converter = tf.contrib.lite.TFLiteConverter.from_session(sess, [x, k], [y])
            tflite_model = converter.convert()

            z = sess.run(y, feed_dict={x: xx, k: kk})

        # Load TFLite model and allocate tensors.
        interpreter = tf.contrib.lite.Interpreter(model_content=tflite_model)
        interpreter.allocate_tensors()

        # Get input and output tensors.
        input_details = interpreter.get_input_details()
        output_details = interpreter.get_output_details()

        interpreter.set_tensor(input_details[0]['index'], xx)
        interpreter.set_tensor(input_details[1]['index'], kk)
        interpreter.invoke()

        output_data = interpreter.get_tensor(output_details[0]['index'])

        assert output_data.shape == OUT_SHAPE

        errs[yf, xf] = np.sqrt(np.sum(np.abs(output_data  - z) ** 2) / z.size)


plt.figure(figsize=(6, 6))
plt.imshow(errs, extent=[0.5, 10.5, 0.5, 10.5])
plt.xlabel("kernel x size")
plt.ylabel("kernel y size")
plt.title("L2 diff in tf/tflite output conv2d_transpose")
plt.colorbar()
plt.savefig("out.png")
#plt.show()
```

### CODE REVIEW

**Comment by perfinion** [comment]
The outputs of the snipped above showing the error between TF and TFLite:

broken
fixed

**Review by jdduke — COMMENTED** [review]
Is there a test we can update to confirm? In transpose_conv_test.cc?

**Comment by perfinion** [comment]
> Is there a test we can update to confirm? In transpose_conv_test.cc?

hmm I havent looked, but something with a non-square padding should trigger it i think.

**Review by jdduke — COMMENTED** [review]
It looks like this resolves the problem for this transpose configuration, but it seems to cause some failures with our existing tranpose_conv tests (bazel test //tensorflow/lite/testing:zip_test_transpose_conv). I'm investigating now.

**Comment by jdduke** [comment]
We have a general internal fix for this, should land in the next day or two.

**Comment by perfinion** [comment]
Looks like this was fixed in https://github.com/tensorflow/tensorflow/commit/c33db572fe31cd48432dab84e17434359786bac8
Thanks! :)

**Comment by perfinion** [comment]
Re-opening. This should be cherry-picked into 1.13

**Comment by aselle** [comment]
Merged https://github.com/tensorflow/tensorflow/pull/25266.
